lib/dependency_validator.dart

```diff
@@ -23,15 +25,25 @@ export 'src/constants.dart' show commonBinaryPackages;
 
 /// Check for missing, under-promoted, over-promoted, and unused dependencies.
 void run({
-  List<String> excludedDirs = const [],
-  bool fatalDevMissing = true,
-  bool fatalMissing = true,
-  bool fatalOverPromoted = true,
-  bool fatalPins = true,
-  bool fatalUnderPromoted = true,
-  bool fatalUnused = true,
-  List<String> ignoredPackages = const [],
+  List<Glob> excludes,
+  bool fatalDevMissing,
+  bool fatalMissing,
+  bool fatalOverPromoted,
+  bool fatalPins,
+  bool fatalUnderPromoted,
+  bool fatalUnused,
+  List<String> ignoredPackages,
 }) {
+  final config = PubspecDepValidatorConfig.fromYaml(
+          File('pubspec.yaml').readAsStringSync())
+      .dependencyValidator;
+  excludes ??= [];
+  excludes.addAll(config?.exclude?.map((s) => Glob(s)) ?? []);
```

Comment:
We should wrap this in a try-catch in case one of the string values in the yaml config is not a valid Glob